data/transactions/logic/jsonspec.md

```diff
@@ -0,0 +1,110 @@
+## JSON Spec
+
+A valid JSON text must follow the grammar defined in <a href="https://www.rfc-editor.org/rfc/rfc7159.html">RFC7159</a>
+
+Additional specifications used by **json_ref** that are extensions to RFC7159 grammar are listed below.
+
+### File Encoding
+- Only utf-8 encoded are accepted.
+- Byte order mark(BOM),"\uFEFF", is not allowed at the beginning of a JSON text,
+  it is treated as an error
+- Raw non-unicode characters not accepted
+#### Invalid JSON text
+```json
+\uFEFF{"key0": 1}
+```
+```json
+{"key0": "\uFF"}
+```
+### Object
+#### duplicate key
+Duplicate key at top level is treated as error and ignored when it is in nested object. 
+#### Invalid JSON text
+```json
+{"key0": 1,"key0": 2}
+```
+#### Valid JSON text
```

Comment:
I hesitate to call this "valid" even if it technically is for the purposes of the opcode. Maybe instead `Acceptable JSON text` ...